automation/src/test/java/com/microsoft/identity/common/test/automation/AcquireTokenSilentForceRefreshTest.java

```diff
@@ -0,0 +1,138 @@
+import java.util.Collection;
+
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+
+import static net.serenitybdd.screenplay.GivenWhenThen.givenThat;
+import static net.serenitybdd.screenplay.GivenWhenThen.seeThat;
+import static net.serenitybdd.screenplay.GivenWhenThen.then;
+import static net.serenitybdd.screenplay.GivenWhenThen.when;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Test case : https://identitydivision.visualstudio.com/IDDP/_workitems/edit/98555
+ */
+
+@RunWith(SerenityParameterizedRunner.class)
+public class AcquireTokenSilentForceRefreshTest {
+
+    @TestData
+    public static Collection<Object[]> FederationProviders(){
+
+
+        return Arrays.asList(new Object[][]{
+                {"ADFSv2"}//,
+                /*)
+                {"ADFSv3"},
+                {"ADFSv4"},
+                {"PingFederate"},
+                {"Shibboleth"}
+                */
+
+        });
+
+    }
+
+    @Steps
+    AcquireTokenSilent acquireTokenSilent;
+
+    @Steps
+    AcquireToken acquireToken;
+
+    @Steps
+    ClickDone clickDone;
+
+    static AppiumDriverLocalService appiumService = null;
+
+    @Managed(driver="Appium")
+    WebDriver hisMobileDevice;
+
+    @BeforeClass
+    public static void startAppiumServer() throws IOException {
+        appiumService = AppiumDriverLocalService.buildDefaultService();
+        appiumService.start();
+    }
+
+    @AfterClass
+    public static void stopAppiumServer() {
+        appiumService.stop();
+    }
+
+    private User james;
+    private String federationProvider;
+
+    public AcquireTokenSilentForceRefreshTest(String federationProvider){
+        this.federationProvider = federationProvider;
+    }
+
+    @Before
+    public void jamesCanUseAMobileDevice(){
+        TestConfigurationQuery query = new TestConfigurationQuery();
+        query.federationProvider = this.federationProvider;
+        query.isFederated = true;
+        query.userType = "Member";
+        james = getUser(query);
+        james.can(BrowseTheWeb.with(hisMobileDevice));
+    }
+
+    private static User getUser(TestConfigurationQuery query){
+
+        Scenario scenario = Scenario.GetScenario(query);
+
+        User newUser = User.named("james");
+        newUser.setFederationProvider(scenario.getTestConfiguration().getUsers().getFederationProvider());
+        newUser.setTokenRequest(scenario.getTokenRequest());
+        //newUser.getTokenRequest().setRedirectUri("msauth://com.microsoft.aad.automation.testapp.adal/1wIqXSqBj7w%2Bh11ZifsnqwgyKrY%3D");
+        newUser.setSilentTokenRequest(scenario.getSilentTokenRequest());
+        //newUser.getSilentTokenRequest().setRedirectUri("msauth://com.microsoft.aad.automation.testapp.adal/1wIqXSqBj7w%2Bh11ZifsnqwgyKrY%3D");
+        newUser.setCredential(scenario.getCredential());
+
+        return newUser;
+    }
+
+
+    @Test
+    public void should_be_able_to_acquire_token_and_then_acquire_silent_with_force_refresh() {
```

Comment:
Is the snake_case something style driven for these tests? Otherwise, this is a departure from the typical camelCase

Reply:
Gotcha 👍